modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -305,6 +305,18 @@ The function imreadmulti loads a specified range from a multi-page image from th
 */
 CV_EXPORTS_W bool imreadmulti(const String& filename, CV_OUT std::vector<Mat>& mats, int start, int count, int flags = IMREAD_ANYCOLOR);
 
+struct CV_EXPORTS_W_SIMPLE AnimationInfo
+{
+    CV_PROP_RW int loop_count;
+    CV_PROP_RW uint32_t bgcolor;
+    CV_PROP_RW int frame_count;
+    CV_PROP_RW std::vector<int> timestamps;
+    CV_PROP_RW std::vector<Mat> frames;
+};
+
+//CV_EXPORTS_W bool imreadanimation(const String& filename, InputArrayOfArrays mats, CV_OUT AnimationInfo& animinfo);
+
+CV_EXPORTS_W bool imreadanimation(const String& filename, CV_OUT AnimationInfo& animinfo);
```

Comment:
@vrabaud it is still experimental but what is your opinion about `bool imreadanimation(const String& filename, CV_OUT AnimationInfo& animinfo)`